linera-views/src/scylla_db.rs

```diff
@@ -474,197 +474,40 @@ impl DirectWritableKeyValueStore<ScyllaDbContextError> for ScyllaDbStoreInternal
-        let session = SessionBuilder::new()
-            .known_node(store_config.uri.as_str())
-            .build()
-            .await?;
-        Self::test_table_existence(&session, &store_config.table_name).await
-    }
-
-    /// Tests if a table is present in the database or not
-    pub async fn test_table_existence(
-        session: &Session,
-        table_name: &String,
-    ) -> Result<bool, ScyllaDbContextError> {
-        // We check the way the test can fail. It can fail in different ways.
-        let query = format!("SELECT dummy FROM kv.{} ALLOW FILTERING", table_name);
+impl AdminKeyValueStore<ScyllaDbContextError> for ScyllaDbStoreInternal {
+    type Config = ScyllaDbStoreConfig;
 
-        // Execute the query
-        let result = session.query(query, &[]).await;
-
-        // The missing table translates into a very specific error that we matched
-        let miss_msg1 = format!("unconfigured table {}", table_name);
-        let miss_msg1 = miss_msg1.as_str();
-        let miss_msg2 = "Undefined name dummy in selection clause";
-        let miss_msg3 = "Keyspace kv does not exist";
-        let Err(error) = result else {
-            // If ok, then the table exists
-            return Ok(true);
-        };
-        let test = match &error {
-            QueryError::DbError(db_error, msg) => {
-                if *db_error != DbError::Invalid {
-                    false
-                } else {
-                    msg.as_str() == miss_msg1
-                        || msg.as_str() == miss_msg2
-                        || msg.as_str() == miss_msg3
-                }
-            }
-            _ => false,
-        };
-        if test {
-            Ok(false)
-        } else {
-            Err(ScyllaDbContextError::ScyllaDbQueryError(error))
-        }
-    }
-
-    fn is_allowed_name(table_name: &str) -> bool {
-        !table_name.is_empty()
-            && table_name.len() <= 48
-            && table_name
-                .chars()
-                .all(|c| c.is_ascii_alphanumeric() || c == '_')
-    }
-
-    /// Creates a table, the keyspace might or might not be existing.
-    ///
-    /// For the table itself, we return true if the table already exists
-    /// and false otherwise.
-    /// If `stop_if_table_exists` is true then the existence of a table
-    /// triggers an error.
-    ///
-    /// This is done since if `new` is called two times, `test_table_existence`
-    /// might be called two times which would lead to `create_table` called
-    /// two times.
-    async fn create_table(
-        session: &Session,
-        table_name: &String,
-        stop_if_table_exists: bool,
-    ) -> Result<bool, ScyllaDbContextError> {
-        if !Self::is_allowed_name(table_name) {
+    async fn connect(config: &Self::Config, namespace: &str) -> Result<Self, ScyllaDbContextError> {
+        if !Self::is_allowed_name(namespace) {
             return Err(ScyllaDbContextError::InvalidTableName);
         }
```

Comment:
You could use `ensure!` whenever you check conditions.

Reply:
I took a different route and instead used `check_namespace` for testing correctness. It is still a one liner.